cmd/harbor/root/config/delete.go

```diff
@@ -0,0 +1,244 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package config
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/goharbor/harbor-cli/pkg/utils"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var deleteAll bool
+
+// DeleteConfigItemCommand creates the 'harbor config delete' subcommand,
+// allowing you to do: harbor config delete <item>
+func DeleteConfigItemCommand() *cobra.Command {
+	var credentialName string
+
+	cmd := &cobra.Command{
+		Use:   "delete <item>",
+		Short: "Delete (clear) a specific config item",
+		Example: `
+  # Clear the current credential's password
+  harbor config delete credentials.password
+
+  # Clear a specific credential's password using --name
+  harbor config delete credentials.password --name admin@http://demo.goharbor.io
+
+  # Clear all credentials and the current credential
+  harbor config delete --all
```

Comment:
can we make  the flag `--current` instead of `--all` since all would mean delete all of the credentials.
```suggestion
  # Clear the current credential
  harbor config delete --all
```

Reply:
BTW can we just set to delete the current config with both username and password by default.

Reply:
What do you mean with `by default`? It is implemented such that the `--current` flag deletes the whole credentials list item. Is that what you mean or can you give me an example?